src/rpc/blockchain.cpp

```diff
@@ -606,6 +606,8 @@ static std::vector<uint8_t> GetRawBlockChecked(BlockManager& blockman, const CBl
         LOCK(cs_main);
         if (blockman.IsBlockPruned(blockindex)) {
             throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+        } else if (!(blockindex.nStatus & BLOCK_HAVE_DATA)) {
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (not fully downloaded)");
         }
```

Comment:
Here and in `rest.cpp`: Structuring it like this should save us the `IsBlockPruned()` call in the normal case (untested). `IsBlockPruned()` checks the same condition plus something extra so this should not change behavior.

```suggestion
        if (!(blockindex.nStatus & BLOCK_HAVE_DATA)) {
            if (blockman.IsBlockPruned(blockindex) {
                throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
            }
            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (not fully downloaded)");
        }
```

Reply:
done. Performance doesn't seem that important, but I also think it's a bit clearer that way.